### ISSUE TITLE
beta.4

### ISSUE THREAD

**Issue by jpudysz** [jpudysz/react-native-unistyles#13]
- [x] Handle transform types to pass media queries
- [x] improve TypeScript typings
- [x] rename `createStyles` to `createStyleSheet` - to be dicussed
- [x] improve docs with themes and variants
- [x] double check other nested styles like `shadowOffset`

**Comment by atanaskanchev** [comment]
I have this ts issue with applying shadows with a dynamic function in beta 3 (not sure if it's fixed in b4)

- theme

```
shadow: {
  $none: {},
  $xs: {
    shadowOffset: {
      width: 0,
      height: 1
    },
    shadowOpacity: 0.18,
    shadowRadius: 1.0,
    shadowColor: 'black',
    elevation: 1
  }
}

```

- createStyles

```tsx
const stylesheet = createStyles((theme) => ({
   shadow: (shadow: Shadow) => theme.shadow[shadow],
})
```

- component
```tsx
 <TouchableOpacity style={[styles.shadow(shadow)]}
```

- ts error on `styles.shadow(shadow)`
```
No overload matches this call.
  Overload 1 of 2, '(props: TouchableOpacityProps | Readonly<TouchableOpacityProps>): TouchableOpacity', gave the following error.
    Type 'unknown' is not assignable to type 'Falsy | ViewStyle | RegisteredStyle<ViewStyle> | RecursiveArray<Falsy | ViewStyle | RegisteredStyle<ViewStyle>> | readonly (Falsy | ... 1 more ... | RegisteredStyle<...>)[]'.
  Overload 2 of 2, '(props: TouchableOpacityProps, context: any): TouchableOpacity', gave the following error.
    Type 'unknown' is not assignable to type 'Falsy | ViewStyle | RegisteredStyle<ViewStyle> | RecursiveArray<Falsy | ViewStyle | RegisteredStyle<ViewStyle>> | readonly (Falsy | ... 1 more ... | RegisteredStyle<...>)[]'.ts(2769)
```

**Comment by jpudysz** [comment]
@atanaskanchev - it should work in beta.4 - I improved types a lot

**Comment by atanaskanchev** [comment]
@jpudysz still an issue with b4
![image](https://github.com/jpudysz/react-native-unistyles/assets/4958587/5cff1799-f021-4d9e-b3d6-19681b546a70)

**Comment by jpudysz** [comment]
@atanaskanchev it's because `$none` is invalid (empty object):

You can quickly fix it by either removing $none or adding any style:

![image](https://github.com/jpudysz/react-native-unistyles/assets/9088288/4f90853f-0992-4eb8-a698-f666eca63d33)

**Comment by jpudysz** [comment]
Nevertheless, if `StyleSheet` supports empty objects, then `react-native-unistyles` should too. I will add it to beta.5

**Comment by atanaskanchev** [comment]
Ah, thanks, makes sense! The error is gone if I remove the empty object. Thanks for the quick turnaround though! I've started to build an app with the library and so far I'm a happy camper (after jumping back and forth between nativewind, twrnc, tamagui and fast-styles :-)

**Comment by jpudysz** [comment]
Thank you, @atanaskanchev; that means a lot to me!
Spotting such things improves the lib a lot 🙏

**Comment by jpudysz** [comment]
Fixed in #18
